src/Combinatorics/Matroids/ChowRings.jl

```diff
@@ -0,0 +1,218 @@
+    Flats = flats(M)
+    number_flats = length(Flats)
+    number_flats >= 2 || error("matroid has to few flats")
+    proper_flats = Flats[2:number_flats-1]
+
+    #construct polynomial ring and extract variables
+    if ring==nothing
+        # create variable names, indexed by the proper flats of M
+        var_names = ["x_{" * join(S, ",") * "}" for S in proper_flats]
+        if extended
+            #add the variables for the simplicial generators
+            var_names = [var_names; ["h_{" * join(S, ",") * "}" for S in proper_flats]]
+        end
+        ring, vars = PolynomialRing(QQ, var_names, cached=false)
+    else
+        nvars(ring) == length(proper_flats) || error("the ring has the wrong number of variables")
+        vars = gens(ring)
+    end
+
+    #construct ideal and quotient
+    I = linear_relations(ring, proper_flats, vars, M)
+    J = quadratic_relations(ring, proper_flats, vars)
+    Ex = elem_type(ring)[]
+    if extended
+        Ex = relations_extended_ring(ring, proper_flats, vars)
+    end
+    chow_modulus = ideal(ring, vcat(I, J, Ex))
+    chow_ring, projection = quo(ring, chow_modulus)
+    return chow_ring
+end
+
+function linear_relations(ring::MPolyRing, proper_flats::GroundsetType, vars::Vector, M::Matroid)
+    alpha = zero(ring)
+    relations = elem_type(ring)[]
+    for i in M.groundset
+        poly = ring(0)
+        for index in findall(issubset([i],F) for F in proper_flats)
+            poly+= vars[index]
+        end
+        if i==M.groundset[1]
+            alpha = poly
+        else
+            push!(relations, alpha-poly)
+        end
+    end
+    return relations
+end
+
+function quadratic_relations(ring::MPolyRing, proper_flats::GroundsetType, vars::Vector)
+    relations = elem_type(ring)[]
+    for i in 1:length(proper_flats)
+        F = proper_flats[i]
+        for j in 1:i-1
+            G = proper_flats[j]
+            if !issubset(F,G) && !issubset(G,F)
+                push!(relations, vars[i]*vars[j])
+            end
+        end
+    end
+    return relations
+end
+
+function relations_extended_ring(ring::MPolyRing, proper_flats::GroundsetType, vars::Vector)
+    relations = elem_type(ring)[]
+    s = length(proper_flats)
+    # h_E = alpha = -x_E
+    poly = ring(0)
+    for index in findall(issubset([1],F) for F in proper_flats)
+        poly+= vars[index]
+    end
+    push!(relations, poly-vars[2*s+1])
+
+    # h_F = h_E - sum x_G where G is a proper flat containing F
+    for i in 1:s
+        F = proper_flats[i]
+        poly = ring(0)
+        for index in findall(issubset(F,G) for G in proper_flats)
+            poly+= vars[index]
+        end
+        push!(relations, poly+vars[s+i]-vars[2*s+1]) #add h_F and subtract h_E
+    end
+    return relations
+end
+
+@doc Markdown.doc"""
+    augmented_chow_ring(M::Matroid)
+
+Return an augmented Chow ring of a matroid. As described in [BHMPW20](@cite). 
+
+Implemented by Fedor Glazov
```

Comment:
```suggestion
```
This should be a comment in the source and not the docstring.

Reply:
done